vesuvius-c.h

```diff
@@ -4482,6 +4542,58 @@ zarr_metadata vs_zarr_parse_zarray(char *path) {
   free(buf);
   return metadata;
 }
+
+chunk* vs_zarr_read_chunk(char* path, zarr_metadata metadata) {
```

Comment:
should probably document in the header that the calling function is responsible for cleaning up allocated memory

Reply:
I'll go confirm, but I think it's always the case that the caller is responsible for `free`ing allocated memory by simply calling `free` if the pointer is a primitive type, otherwise calling `_free` e.g. `chunk_free(chunk* p)` which will free any embedded `malloc`ed code _and_ free the struct pointer itself.

I'll make a note of this in the comment block I have buried in the code and move it to the top of the header.

Reply:
addressed, see the top of vesuvius-c.h for notes on memory management